[brianchoon] iP

brianchoon opened the pull request:
# Devon

> “Simplicity is the ultimate sophistication.” – Leonardo da Vinci

Devon, your personalized assistant, has been updated to enhance user experience with new features :fire:. Here's what has changed:

- **More intuitive commands**
- **Improved response time**
- *Enhanced* data handling capabilities

## Steps to use the updated version:

1. Download the latest version from [this website](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. Run the application
3. Start interacting with Devon to explore the new features.

### Features:

- [x] Respond to custom commands, including, but not limited to: todo, deadlines, events
- [x] Manage completeness of tasks
- [ ] AI-based suggestions (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
    Devon bot = new Devon();
    bot.start();
});

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

icantshootfloorballs committed:
Rename main class and add greeting and goodbye

icantshootfloorballs committed:
Add Scanner

icantshootfloorballs committed:
Add indentation for bot messages

icantshootfloorballs committed:
Add echo

icantshootfloorballs committed:
Add array to store tasks

icantshootfloorballs committed:
Add list command

icantshootfloorballs committed:
Implement Task class

icantshootfloorballs committed:
Add support to mark done/undone

icantshootfloorballs committed:
Refactor toString

icantshootfloorballs committed:
Add support for different tasks

icantshootfloorballs committed:
Add support for commands for each task type

icantshootfloorballs committed:
Implement I/O redirection

icantshootfloorballs committed:
Implement error handling in main code

icantshootfloorballs committed:
Implement ArrayList

icantshootfloorballs committed:
Add support for deleting tasks

icantshootfloorballs committed:
Implement enums for commands

icantshootfloorballs committed:
Add support for local storage of tasks

icantshootfloorballs committed:
Merge branch-Level-7 into master

icantshootfloorballs committed:
Implement Date and Time support

icantshootfloorballs committed:
Merge branch-Level-8 into master

icantshootfloorballs committed:
Implement TaskList class

icantshootfloorballs committed:
Add Storage class

icantshootfloorballs committed:
Refactor printList function

icantshootfloorballs committed:
Fix bug in printList

icantshootfloorballs committed:
Refactor task functions

icantshootfloorballs committed:
Implement parser class

icantshootfloorballs committed:
Implement UI class

icantshootfloorballs committed:
Merge branch-A-MoreOOP into master

icantshootfloorballs committed:
Create package for devon files

icantshootfloorballs committed:
Merge branch-A-Packages into master

icantshootfloorballs committed:
Merge add-gradle-support into master

icantshootfloorballs committed:
Add Gradle support for main

icantshootfloorballs committed:
Merge branch-A-Gradle into master

icantshootfloorballs committed:
Add JUnit tests for Todo and Deadline

icantshootfloorballs committed:
Merge branch-A-JUnit into master

icantshootfloorballs committed:
Set up shadowJar

icantshootfloorballs committed:
Merge branch-A-Jar into master

icantshootfloorballs committed:
Add JavaDoc for all methods in all classes

icantshootfloorballs committed:
Edit code to comply with coding standard

icantshootfloorballs committed:
Implement search feature for task descriptions

icantshootfloorballs committed:
Merge branch-A-JavaDoc into master

icantshootfloorballs committed:
Merge branch-A-CodingStandard into master

icantshootfloorballs committed:
Merge branch-Level-9 into master

icantshootfloorballs committed:
Add JavaDoc for all new methods

maahir-garg commented on src/main/java/devon/Devon.java:
```diff
@@ -0,0 +1,318 @@
+package devon;
+
+import java.io.*;
```
avoid wildcard imports

maahir-garg commented on src/main/java/devon/Devon.java:
```diff
@@ -0,0 +1,318 @@
+            Command command = Command.fromStringToEnum(parser.extractCommand(input));
+
+            try {
+                switch (command) {
+                    case BYE:
+                        return;
+                    case LIST:
+                        printList();
+                        break;
+                    case MARK:
+                        markAction(input);
+                        break;
+                    case UNMARK:
+                        unmarkAction(input);
+                        break;
+                    case TODO:
+                        todoAction(input);
+                        break;
+                    case DEADLINE:
+                        deadlineAction(input);
+                        break;
+                    case EVENT:
+                        eventAction(input);
+                        break;
+                    case DELETE:
+                        deleteAction(input);
+                        break;
+                    case FIND:
+                        findAction(input);
+                        break;
+                    default:
+                        unknownAction();
+                        break;
```
avoid indentation for switch case

maahir-garg commented on src/main/java/devon/Storage.java:
```diff
@@ -0,0 +1,74 @@
+    protected static final String DB_PATH = String.valueOf(Paths.get(Storage.DIRECTORY_PATH, "devon_tasks.txt"));
+    protected static final String DB_DELIMITER = "\\|";
+
+    protected static final DateTimeFormatter DATE_TIME_FORMATTER_FOR_EXTERNAL_INPUT = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    protected static final DateTimeFormatter DATE_TIME_FORMATTER_FOR_DB = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm");
```
Avoid characters more than 120

maahir-garg reviewed (commented):
Overall, well done.. try thinking If we can divide the classes into more packages like commands and tasks

IanCheah commented on src/main/java/devon/Devon.java:
```diff
@@ -0,0 +1,318 @@
+            Command command = Command.fromStringToEnum(parser.extractCommand(input));
+
+            try {
+                switch (command) {
```
The indentation of the cases should be of the same level as the switch 😅

IanCheah commented on src/test/java/devon/DeadlineTest.java:
```diff
@@ -0,0 +1,48 @@
+
+public class DeadlineTest {
+    @Test
+    public void dbReadableFormat_undone_success() {
```
good use of underscore for test methods 😄

IanCheah commented on src/main/java/devon/DevonInvalidTaskNumberException.java:
```diff
@@ -0,0 +1,20 @@
+ * This exception is a specific type of {@link DevonException} used to indicate errors
+ * related to task number handling.
+ */
+public class DevonInvalidTaskNumberException extends DevonException {
```
good use of camel case for class name. 😄

IanCheah commented on src/main/java/devon/Devon.java:
```diff
@@ -0,0 +1,318 @@
+     * @throws DevonInvalidDateTimeException If the date/time format is invalid.
+     */
+    private void deadlineAction(String input) throws DevonInvalidDeadlineException, DevonInvalidDateTimeException {
+        String[] contents = parser.extractDeadline(input);
```
Good use of plural contents 😄

icantshootfloorballs committed:
Fix indentation for switch statements

icantshootfloorballs committed:
Implement checkstyle

icantshootfloorballs committed:
Merge branch-A-Checkstyle into master

icantshootfloorballs committed:
Fix all checkstyle errors

icantshootfloorballs committed:
Update build.gradle file

icantshootfloorballs committed:
Set up basic template GUI

icantshootfloorballs committed:
Rename Duke to Devon

icantshootfloorballs committed:
Implement one-way chat feature

icantshootfloorballs committed:
Implement two-way chat feature

icantshootfloorballs committed:
Use FXML

icantshootfloorballs committed:
Complete implementation of GUI

icantshootfloorballs committed:
Merge branch-Level-10 into master

icantshootfloorballs committed:
Fix checkstyle errors

icantshootfloorballs committed:
Remove extra backslash

icantshootfloorballs committed:
Main.java: Replace Duke reference with Devon

Update the comment in Main.java to correctly refer to the class as Devon instead of Duke.

The previous comment referred to Duke, which could lead to confusion when reading the code. Changing this ensures that the codebase accurately reflects the current state of the project.

No other functionality is affected by this change, as it is purely a documentation update.

icantshootfloorballs committed:
Add assertions to ensure key assumptions hold in Devon bot

Added assertions in deadlineAction and eventAction to verify non-null command parsing before proceeding with operations. These assertions improve error handling and document important invariants, enhancing code reliability and readability.

icantshootfloorballs committed:
Add assertions to ensure key assumptions hold in Devon bot

Added assertions in deadlineAction and eventAction to verify non-null command parsing before proceeding with operations. These assertions improve error handling and document important invariants, enhancing code reliability and readability.

icantshootfloorballs committed:
Remove redundant variables

icantshootfloorballs committed:
Add whitespace for readability

icantshootfloorballs committed:
Remove unused variables

icantshootfloorballs committed:
Modify constant variables to be final

brianchoon committed:
Merge pull request #1 from brianchoon/branch-A-Assertions

Add Assertions to Document Key Assumptions in the Devon Bot

icantshootfloorballs committed:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

brianchoon committed:
Merge pull request #2 from brianchoon/branch-A-CodeQuality

Improve Code Quality

icantshootfloorballs committed:
Add alias commands for todo, deadline, event

icantshootfloorballs committed:
Merge branch-C-FriendlierSyntax into master

icantshootfloorballs committed:
Customise user and Devon's display pictures

icantshootfloorballs committed:
Update README.md to declare AI usage

icantshootfloorballs committed:
Add screenshot of product

icantshootfloorballs committed:
Complete user guide
